addon/components/file-upload.ts

```diff
@@ -105,21 +104,17 @@ export default class FileUploadComponent extends Component<FileUploadArgs> {
       return this.args.queue;
     }
 
-    if (this.args.name) {
-      return this.fileQueue.findOrCreate(this.args.name);
-    }
-
-    return this.fileQueue.findOrCreate(DEFAULT_QUEUE);
+    return this.fileQueue.findOrCreate(this.args.name ?? DEFAULT_QUEUE);
   }
 
   get for() {
     return this.args.for || `file-input-${uuid.short()}`;
   }
 
-  @action
-  attachListeners() {
+  bindListeners = modifier(() => {
     this.queue.addListener(this);
-  }
+    return () => this.queue.removeListener(this);
+  });
```

Comment:
Is there any typing information I should include here? I don't think the modifier needs to accept arguments since the whole object (this) is what's bound/unbound.

Reply:
As there are no args involved... enjoy the clean look :)